libs/stringCore/include/stringCore/msostringTraits.h

```diff
@@ -0,0 +1,57 @@
+/*-------------------------------------------------------------------------
+  msoStringTraits.h
+
+  Copyright (c) 1994 Microsoft Corporation
+
+  Functions related to string and character characteristics.
+-------------------------------------------------------------------------*/
+#pragma once
+#ifndef LIBLET_ROOT_MSOSTRINGTRAITS_H
+#define LIBLET_ROOT_MSOSTRINGTRAITS_H
+#include <compilerAdapters/functionDecorations.h>
+
+#define MsoFIsLowerFast(ch) ((unsigned)((ch) - 'a') <= 'z' - 'a')
+#define ASCII_TOUPPER(x) (FBetween(x, 'a', 'z') ? (x) - ('a' - 'A') : (x))
+
+#define UTF8_2_MAX 0x07ff // max UTF8 2-byte sequence (32 * 64 = 2048)

```

Comment:
>UTF8_2_MAX [](http://example.com/codeflow?start=8&length=10)

All these macros look quite ugly.
Do we need them?
Can we replace them with constexpr inline variables and functions?